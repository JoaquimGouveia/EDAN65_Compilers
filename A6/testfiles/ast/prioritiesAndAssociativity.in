int a() {
    int x = 5 * 3 + 2;
    int z = 10 - 6 / 2;
    int a = (2 + 3) * 4;
    int b = 8 - 3 - 2;
    int c = 2 * 3 * 4;
    int d = 5 + 6 - 2 + 1;
    int e = 10 / 2 * 3;
    int f = (5 + 2) * (8 - 3);
    int g = ((3 + 2) * (2 + 3)) - 5;
    int h = 10 % 3 + 4;
    int j = -5 * 3;
    int k = -(5 + 3) * 2;
}