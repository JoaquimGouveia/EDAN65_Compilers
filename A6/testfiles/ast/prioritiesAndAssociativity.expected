Program
  List
    FuncDecl
      IdDecl ID="a"
      List
      Block
        List
          VarDeclStmt
            IdDecl ID="x"
            Opt
              Add
                Mul
                  Numeral NUMERAL="5"
                  Numeral NUMERAL="3"
                Numeral NUMERAL="2"
          VarDeclStmt
            IdDecl ID="z"
            Opt
              Sub
                Numeral NUMERAL="10"
                Div
                  Numeral NUMERAL="6"
                  Numeral NUMERAL="2"
          VarDeclStmt
            IdDecl ID="a"
            Opt
              Mul
                Add
                  Numeral NUMERAL="2"
                  Numeral NUMERAL="3"
                Numeral NUMERAL="4"
          VarDeclStmt
            IdDecl ID="b"
            Opt
              Sub
                Sub
                  Numeral NUMERAL="8"
                  Numeral NUMERAL="3"
                Numeral NUMERAL="2"
          VarDeclStmt
            IdDecl ID="c"
            Opt
              Mul
                Mul
                  Numeral NUMERAL="2"
                  Numeral NUMERAL="3"
                Numeral NUMERAL="4"
          VarDeclStmt
            IdDecl ID="d"
            Opt
              Add
                Sub
                  Add
                    Numeral NUMERAL="5"
                    Numeral NUMERAL="6"
                  Numeral NUMERAL="2"
                Numeral NUMERAL="1"
          VarDeclStmt
            IdDecl ID="e"
            Opt
              Mul
                Div
                  Numeral NUMERAL="10"
                  Numeral NUMERAL="2"
                Numeral NUMERAL="3"
          VarDeclStmt
            IdDecl ID="f"
            Opt
              Mul
                Add
                  Numeral NUMERAL="5"
                  Numeral NUMERAL="2"
                Sub
                  Numeral NUMERAL="8"
                  Numeral NUMERAL="3"
          VarDeclStmt
            IdDecl ID="g"
            Opt
              Sub
                Mul
                  Add
                    Numeral NUMERAL="3"
                    Numeral NUMERAL="2"
                  Add
                    Numeral NUMERAL="2"
                    Numeral NUMERAL="3"
                Numeral NUMERAL="5"
          VarDeclStmt
            IdDecl ID="h"
            Opt
              Add
                Mod
                  Numeral NUMERAL="10"
                  Numeral NUMERAL="3"
                Numeral NUMERAL="4"
          VarDeclStmt
            IdDecl ID="j"
            Opt
              Mul
                Neg
                  Numeral NUMERAL="5"
                Numeral NUMERAL="3"
          VarDeclStmt
            IdDecl ID="k"
            Opt
              Mul
                Neg
                  Add
                    Numeral NUMERAL="5"
                    Numeral NUMERAL="3"
                Numeral NUMERAL="2"
